scripts/org-setup.mjs

```diff
@@ -1,58 +1,80 @@
-import 'zx/globals'
-import { writeFile } from 'node:fs/promises'
-import { join } from 'node:path'
-import { EOL } from 'node:os'
-
-if (!process.env.SF_HUB_USERNAME || !process.env.SF_OAUTH2_JWT_KEY || !process.env.SF_OAUTH2_CLIENT_ID) {
-  throw new Error('Set SF_HUB_USERNAME, SF_OAUTH2_JWT_KEY and SF_OAUTH2_CLIENT_ID env vars.')
-}
-
-$.verbose = false
-
-const hubOpts = {
-  username: process.env.SF_HUB_USERNAME,
-  clientId: process.env.SF_OAUTH2_CLIENT_ID,
-  jwtKeyFile: join(process.cwd(), 'jwtKey.txt')
+import 'zx/globals';
+import { writeFile } from 'node:fs/promises';
+import { join } from 'node:path';
+import { EOL } from 'node:os';
+
+$.verbose = false;
+
+if (
+  process.env.SF_HUB_USERNAME &&
+  process.env.SF_OAUTH2_JWT_KEY &&
+  process.env.SF_OAUTH2_CLIENT_ID
+) {
+  const hubOpts = {
+    username: process.env.SF_HUB_USERNAME,
+    clientId: process.env.SF_OAUTH2_CLIENT_ID,
+    jwtKeyFile: join(process.cwd(), 'jwtKey.txt'),
+  };
+
+  await writeFile(hubOpts.jwtKeyFile, formatJwtKey());
+
+  await $`sf org login jwt --username ${hubOpts.username} --jwt-key-file "${hubOpts.jwtKeyFile}" --set-default-dev-hub --client-id ${hubOpts.clientId}`;
+} else {
+  // ensure there's a default devhub
+  const defaultHub = JSON.parse(await $`sf config get target-dev-hub --json`);
+  console.log(defaultHub);
+
+  if (defaultHub.result.find((res) => res.sucesss && res.value) !== undefined) {
+    throw new Error(
+      'Set SF_HUB_USERNAME, SF_OAUTH2_JWT_KEY and SF_OAUTH2_CLIENT_ID env vars.',
+    );
+  }
```

Comment:
update to make it work with an auth'd hub set as a default global, see:
https://github.com/jsforce/jsforce/blob/cd/revamp-build-scripts/DEVELOPING.md#scratch-org-setup-for-testing

`npx zx scripts/org-setup.mjs` without the hub env vars will look for the default one and fail if there's none.
no change required in CI.

Reply:
yup, added this in the org setup section, thanks!
https://github.com/jsforce/jsforce/pull/1369/commits/b5ba8e7224a457b1b9134115df7476345dcc3834